src/contexts/SurveyProvider.tsx

```diff
@@ -0,0 +1,89 @@
+'use client'
+import {
+  createContext,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react'
+import { usePathname, useSearchParams, useRouter } from 'next/navigation'
+import { SurveyData } from '@/@types/survey'
+import {
+  GenderQuestion,
+  NameQuestion,
+  PetTypeQuestion,
+} from '@/components/Survey'
+
+type SurveyContextSchema = {
+  currentStep: number
+  isFirstQuestion: boolean
+  isLastQuestion: boolean
+  nextStep: () => void
+  prevStep: () => void
+  surveyComponents: Array<(props: PropsWithChildren) => JSX.Element>
```

Comment:
```suggestion
  surveyComponents: Array<(props?: PropsWithChildren) => JSX.Element>
```
Not all have children right